### PR TITLE
docs: add apisix with consul blog

### PR DESCRIPTION
**Pull request by Fabriceli**
Fixes: #[Add issue number here]

Changes:

add apisix with consul blog

Screenshots of the change:

### CODE REVIEW

**Comment by SylviaBABY** [comment]
@Fabriceli thanks for your contribution! Looking forward to this article. 

You can check this part to fix Lint errors
image
image

**Comment by Fabriceli** [comment]
> @Fabriceli thanks for your contribution! Looking forward to this article.
> 
> You can check this part to fix Lint errors image image

Done

File: blog/en/blog/2023/01/18/consul-with-apisix.md [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+title: "How to Integrate API Gateway and Consul? Not Consul K/V"
+authors:
+  - name: "Yihao LI"
+    title: "Author"
+    url: "https://github.com/Fabriceli"
+    image_url: "https://github.com/Fabriceli.png"
+  - name: "Yihao LI"
+    title: "Technical Writer"
+    url: "https://github.com/Fabriceli"
+    image_url: "https://github.com/Fabriceli.png"
```
Comment (SylviaBABY) [inline blog/en/blog/2023/01/18/consul-with-apisix.md]: you can delete this part, only keep "Author". Cause you are the author of this article, and no one else edits this article except you. such as this:(same with ZH)
image

File: blog/zh/blog/2023/01/18/consul-with-apisix.md [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+title: "Apache APISIX 集成原生 Consul 服务发现能力"
+authors:
+  - name: "李奕浩"
+    title: "Author"
+    url: "https://github.com/Fabriceli"
+    image_url: "https://github.com/Fabriceli.png"
+  - name: "李奕浩"
+    title: "Technical Writer"
+    url: "https://github.com/Fabriceli"
+    image_url: "https://github.com/Fabriceli.png"
```
Comment (SylviaBABY) [inline blog/zh/blog/2023/01/18/consul-with-apisix.md]: ditto

File: blog/zh/blog/2023/01/18/consul-with-apisix.md [hunk 3]
```diff
@@ -0,0 +1,271 @@
...
+keywords:
+- API Gateway
+- Consul
+- 服务发现
+- 服务注册
+description: 云原生 API 网关 Apache APISIX 支持基于 Consul 的服务发现。本文讲述了在 Apache APISIX 中实现服务发现和服务注册的全过程及 consul 的相关原理。
```
Comment (SylviaBABY) [inline blog/zh/blog/2023/01/18/consul-with-apisix.md]:
```suggestion
description: 云原生 API 网关 Apache APISIX 支持基于 Consul 的服务发现。本文讲述了在 Apache APISIX 中实现服务发现和服务注册的全过程及 Consul 的相关原理。
```

File: blog/zh/blog/2023/01/18/consul-with-apisix.md [hunk 4]
```diff
@@ -0,0 +1,271 @@
...
+      "CreateIndex": 47,
+      "ModifyIndex": 125
+   }]
+  ```
+
+## 创建路由
+
+### 添加路由
+
+在添加之前需要确认 `X-API-KEY`，属于 APISIX Admin API 的访问 Token，在该示例中，使用默认的即可：`edd1c9f034335f136f87ad84b625c8f1`，这里将 URL 为`/consul/web/*`的请求路由分配到 Consul 的 service_a。
```
Comment (SylviaBABY) [inline blog/zh/blog/2023/01/18/consul-with-apisix.md]:
```suggestion
在添加之前需要确认 `X-API-KEY`，属于 APISIX Admin API 的访问 Token，在该示例中，使用默认的即可：`edd1c9f034335f136f87ad84b625c8f1`，这里将 URL 为 `/consul/web/*` 的请求路由分配到 Consul 的 service_a。
```

File: blog/zh/blog/2023/01/18/consul-with-apisix.md [hunk 5]
```diff
@@ -0,0 +1,271 @@
...
+      "create_time": 1674029322,
+      "priority": 0
+   },
+   "key": "\/apisix\/routes\/1"
+}
+```
+
+## 验证配置结果
+
+通过请求结果可以看到，Apache APISIX 中新增的路由通过 Consul 命中了正确的服务地址，并根据`roundrobin`负载均衡策略请求到两个节点上。
```
Comment (SylviaBABY) [inline blog/zh/blog/2023/01/18/consul-with-apisix.md]:
```suggestion
通过请求结果可以看到，Apache APISIX 中新增的路由通过 Consul 命中了正确的服务地址，并根据 `roundrobin` 负载均衡策略请求到两个节点上。
```

Comment (Fabriceli) [reply blog/zh/blog/2023/01/18/consul-with-apisix.md]: done

File: blog/zh/blog/2023/01/18/consul-with-apisix.md (hunk 4 above)
Comment (Fabriceli) [reply blog/zh/blog/2023/01/18/consul-with-apisix.md]: done

File: blog/zh/blog/2023/01/18/consul-with-apisix.md (hunk 3 above)
Comment (Fabriceli) [reply blog/zh/blog/2023/01/18/consul-with-apisix.md]: done

File: blog/zh/blog/2023/01/18/consul-with-apisix.md (hunk 2 above)
Comment (Fabriceli) [reply blog/zh/blog/2023/01/18/consul-with-apisix.md]: done

File: blog/en/blog/2023/01/18/consul-with-apisix.md (hunk 1 above)
Comment (Fabriceli) [reply blog/en/blog/2023/01/18/consul-with-apisix.md]: done

File: blog/en/blog/2023/01/18/consul-with-apisix.md [hunk 6]
```diff
@@ -0,0 +1,274 @@
...
+keywords:
+- API Gateway
+- Consul
+- Service Discovery
+- Service Register
+description: Apache APISIX supports the Consul service discovery registry. This article will walk you through the process of implementing service discovery and service registry in APISIX.
+tags: [Ecosystem]
+image: https://static.apiseven.com/2022/blog/0818/ecosystem/HashiCorp%20Consul.png
+---
+
```
Comment (yzeng25) [inline blog/en/blog/2023/01/18/consul-with-apisix.md:17]: 1. Need a `<!--truncate-->` here to cut actual contents off from the blog summary page
2. Need a sentence or two to add a summary for this blog.
```suggestion
> This article teaches you how to integrate Consul with Apache APISIX 
<!--truncate-->

```
Now it looks like this, and it is just not right.

image

Comment (Fabriceli) [reply blog/en/blog/2023/01/18/consul-with-apisix.md:17]: done

**Review by yzeng25 — APPROVED** [review]

**Review by SylviaBABY — APPROVED** [review]

**Comment by Fabriceli** [comment]
@SylviaBABY @yzeng25 I have no permission to merge this PR, Could you merge it?

**Pull request merged**